packages/@webex/plugin-meetings/src/media/properties.js

```diff
@@ -195,4 +197,99 @@ export default class MediaProperties {
     this.unsetLocalVideoTrack();
     this.unsetRemoteMedia();
   }
+
+  /**
+   * Waits until ice connection is established
+   *
+   * @returns {Promise<void>}
+   */
+  waitForIceConnectedState() {
+    const isIceConnected = () => (
+      this.peerConnection.iceConnectionState === ICE_STATE.CONNECTED ||
+        this.peerConnection.iceConnectionState === ICE_STATE.COMPLETED
+    );
+
+    if (isIceConnected()) {
+      return Promise.resolve();
+    }
+
+    return new Promise((resolve, reject) => {
+      let timer;
+
+      const iceListener = () => {
+        LoggerProxy.logger.log(`Media:properties#waitForIceConnectedState --> ice state: ${this.peerConnection.iceConnectionState}, conn state: ${this.peerConnection.connectionState}`);
+
+        if (isIceConnected()) {
+          clearTimeout(timer);
+          this.peerConnection.removeEventListener('iceconnectionstatechange', iceListener);
+          resolve();
+        }
+      };
+
+      timer = setTimeout(() => {
+        this.peerConnection.removeEventListener('iceconnectionstatechange', iceListener);
+        reject();
+      }, PC_BAIL_TIMEOUT);
+
+      this.peerConnection.addEventListener('iceconnectionstatechange', iceListener);
```

Comment:
@marcin-bazyl  i am assuming this wont override the event registred in peer connection manager

Reply:
correct, you can call `addEventListener()` multiple times to register multiple listeners

Reply:
actually, peer connection manager doesn't use `addEventListener()` but just sets `oniceconnectionstatechange` directly. This is also fine - it works, I've just checked this by running the sample app with the code from this PR and I could see the "PeerConnectionManager:index#setPeerConnectionEvents" logs in the console when joining meeting and also later when I blocked and unblocked UDP traffic, so it's fine.

Reply:
I've checked it on chrome, firefox and safari